internal/api/grpc/user/v2/integration_test/query_test.go

```diff
@@ -861,6 +923,7 @@ func TestServer_ListUsers(t *testing.T) {
 					orgResp := Instance.CreateOrganization(ctx, fmt.Sprintf("ListUsersResourceowner-%s", gofakeit.AppName()), gofakeit.Email())
 
 					infos := createUsers(ctx, orgResp.OrganizationId, 3, false)
+					request.Queries = []*user.SearchQuery{}
```

Comment:
Perhaps add one more test case filtering for the specific org, so that we also cover the org optimization you did earlier.